azurelinuxagent/common/dhcp.py

```diff
@@ -151,9 +152,15 @@ def send_dhcp_req(self):
         """
         Check if DHCP is available
         """
-        dhcp_available =  self.osutil.is_dhcp_available()
-        if not dhcp_available:
-            logger.info("send_dhcp_req: DHCP not available")
+        dhcp_available = self.osutil.is_dhcp_available()
+        # If user has DHCP disabled for their VM then the dhcp request will fail. The user can configure the agent to
+        # use the hardcoded wire server ip instead.
+        use_dhcp = conf.get_dhcp_discovery_enabled()
```

Comment:
Treating dhcp discovery config the same as when dhcp is unavailable in osutil

Reply:
but if conf.get_dhcp_discovery_enabled() == False, we should not even reach this function

Reply:
Same is true if self.osutil.is_dhcp_available() is False. I followed the same pattern for conf.get_dhcp_discover_enabled(). 
I assumed the is_dhcp_available() check was here in case another dhcp_handler.run() call is added in the future.